nwcsafpps_runner/utils.py

```diff
@@ -588,15 +588,30 @@ def get_xml_outputfiles(path, platform_name, orb, st_time=''):
 
 
 def create_xml_timestat_from_lvl1c(scene, pps_control_path):
-    """From lvl1c file create XML file and return a file list."""
+    """From lvl1c file create XML file and return a file list, v2021."""
     try:
-        return [create_pps_file_from_lvl1c(scene['file4pps'], pps_control_path, "timectrl", ".xml")]
+        txt_time_control = create_pps_file_from_lvl1c(scene['file4pps'], pps_control_path, "timectrl", ".txt")
     except KeyError:
         return []
+    if os.path.exists(txt_time_control):
+        return create_xml_timestat_from_ascii(txt_time_control, pps_control_path) 
+    else:
+        LOG.warning('No XML Time statistics file created!')
+        return []
+
+
+def create_xml_timestat_from_scene(scene, pps_control_path):
+    """From lvl1c file create XML file and return a file list, v2018."""
+    try:
+        txt_time_control = get_time_control_ascii_filename(scene, pps_control_path)
+    except FindTimeControlFileError:
+        LOG.exception('No XML Time statistics file created!')
+        return []
+    return create_xml_timestat_from_ascii(txt_time_control, pps_control_path)
 
 
-def create_product_statistics_from_lvl1c(scene, pps_control_path):
-    """From lvl1c file create product XML files and return a file list."""
+def find_product_statistics_from_lvl1c(scene, pps_control_path):
```

Comment:
I am probably to blame, but the fact that the functions return a list is not clear from reading the name of the file. I wonder if some of it could be broken down in smaller pieces?

Reply:
I think we can to that in a in a separate PR. And soon perhaps we could remove all the v2018 code. That would make it clearer how to refactor I think.

Reply:
When we no longer need version 2018 we can remove the functions get_outputfiles, filter4oldfiles, get_xml_outputfiles, filter4oldfiles, create_xml_timestat_from_scene, get_time_control_ascii_filename, get_time_control_ascii_filename_candidates, get_product_statistics_files I think.